code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -322,12 +324,20 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 
 /obj/item/clothing/mask/cigarette/process()
 	var/turf/location = get_turf(src)
-	var/mob/living/M = loc
+	var/mob/living/M
+	if(inside_item)
+		if(loc)
```

Comment:
There's helpers for that, find holder of type for instance and similar procs to it, there should be like 2 or 3 in unsorted.dm